nixos-modules/valheim.nix

```diff
@@ -61,6 +61,43 @@ in {
       description = lib.mdDoc "Whether to open ports in the firewall.";
     };
 
+    public = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+      description = lib.mdDoc ''
+        Toggles visibility on the Steam server & community lists.
+
+        When not set, defaults to true (visible).
+        Set to false to make the server private.
+      '';
+    };
+
+    logFile = lib.mkOption {
+      type = with lib.types; nullOr str;
+      default = null;
+      example = "/var/lib/valheim/log/valheim-server.log";
+      description = lib.mdDoc ''
+        The path where the server log file will be saved.
+
+        When set, this will redirect all logs from stdout to the specified path.
+        This means that the logs will not be captured by systemd's journal.
+        Leave this option unset to keep console logging enabled.
+
+        Make sure the valheim user has write access to the specified directory, otherwise
+        the valheim service fail to start and exit.
+      '';
+    };
+
+    preset = lib.mkOption {
+      type = with lib.types; nullOr str;
+      default = null;
+      example = "hardcore";
+      description = lib.mdDoc ''
+        The preset world modifier, valid options are
+        "easy", "hard", "hardcore", "casual", "hammer" and "immersive".
+      '';
+    };
```

Comment:
Regarding your comment about using an enum type, you would change the type from `nullOr str` to `nullOr (enum ["easy" "hard" ...])` (replacing `...` with the rest of the enum values), and then remove the list of values from the description.

Reference: [NixOS Manual](https://nixos.org/manual/nixos/stable/#sec-option-declarations-eot)